tests/unit/browser/test_qutescheme.py

```diff
@@ -27,30 +28,72 @@
 from qutebrowser.utils import objreg
 
 
-Dates = collections.namedtuple('Dates', ['yesterday', 'today', 'tomorrow'])
+class TestJavascriptHandler:
+
+    """Test the qute://javascript endpoint."""
+
+    # Tuples of fake JS files and their content.
+    js_files = [
+        ('foo.js', "var a = 'foo';"),
+        ('bar.js', "var a = 'bar';"),
+    ]
+
+    @pytest.fixture(autouse=True)
+    def patch_read_file(self, monkeypatch):
+        """Patch utils.read_file to return few fake JS files."""
+        def _read_file(path, binary=False):
+            """Faked utils.read_file."""
+            assert not binary
+            for filename, content in self.js_files:
+                if path == os.path.join('javascript', filename):
```

Comment:
This will give you `javascript\foo.js` on Windows, but `read_file` gets called with the path from the URL, which is always using forward slashes.

I'll comment with a possible solution in the handler code.